integrationtest/integration_test.go

```diff
@@ -100,7 +100,38 @@ func annotateServiceAccount(t *testing.T, name string, annotations map[string]st
 	}
 }
 
-func setupKubernetesAuth(t *testing.T, boundServiceAccountName string, mountConfigOverride map[string]interface{}, roleConfigOverride map[string]interface{}) (*api.Client, func()) {
+func createPolicy(t *testing.T, name, policy string) func() {
+	t.Helper()
+	// Pick up VAULT_ADDR and VAULT_TOKEN from env vars
+	client, err := api.NewClient(nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = client.Logical().Write(fmt.Sprintf("/sys/policy/%s", name), map[string]interface{}{
+		"policy": policy,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	cleanup := func() {
+		_, err = client.Logical().Delete(fmt.Sprintf("/sys/policy/%s", name))
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	defer func() {
+		if t.Failed() {
+			cleanup()
+		}
+	}()
+
+	return cleanup
```

Comment:
nit: I think you can use t.Cleanup to simplify a bit

```suggestion
	t.Cleanup(func() {
		_, err = client.Logical().Delete(fmt.Sprintf("/sys/policy/%s", name))
		if err != nil {
			t.Fatal(err)
		}
	})
```